### ISSUE TITLE
Test DependencyInstaller script / testInstaller (rhel, gcc) fails with: cannot find -ltcl

### ISSUE THREAD

**Issue by rovinski** [The-OpenROAD-Project/OpenROAD#3343]
### Describe the bug

Running `./etc/DockerHelper.sh create -target=builder -os=centos7 -compiler=gcc` (as well as a few others) results in a failure in the CI with:
```
/usr/bin/ld: cannot find -ltcl
```
The error is reproducible on my local machine using docker.

### Expected Behavior

Successful build

### Environment

```shell
commit b2fabb1
```

### To Reproduce

`./etc/DockerHelper.sh create -target=builder -os=centos7 -compiler=gcc`

### Additional Context

I have tracked down the issue to some kind of bug in the linking of `libtcl.so`

Here is the linker debug output (gcc -v -wl,--verbose):

```
attempt to open /usr/lib/gcc/x86_64-redhat-linux/8/libtcl.so failed
attempt to open /usr/lib/gcc/x86_64-redhat-linux/8/libtcl.a failed
attempt to open /usr/lib/gcc/x86_64-redhat-linux/8/../../../../lib64/libtcl.so failed
attempt to open /usr/lib/gcc/x86_64-redhat-linux/8/../../../../lib64/libtcl.a failed
attempt to open /lib/../lib64/libtcl.so failed
attempt to open /lib/../lib64/libtcl.a failed
attempt to open /usr/lib/../lib64/libtcl.so failed
attempt to open /usr/lib/../lib64/libtcl.a failed
attempt to open /usr/lib/gcc/x86_64-redhat-linux/8/../../../libtcl.so failed
attempt to open /usr/lib/gcc/x86_64-redhat-linux/8/../../../libtcl.a failed
attempt to open //usr/x86_64-redhat-linux/lib64/libtcl.so failed
attempt to open //usr/x86_64-redhat-linux/lib64/libtcl.a failed
attempt to open //usr/lib64/libtcl.so failed
attempt to open //usr/lib64/libtcl.a failed
attempt to open //usr/local/lib64/libtcl.so failed
attempt to open //usr/local/lib64/libtcl.a failed
attempt to open //lib64/libtcl.so failed
attempt to open //lib64/libtcl.a failed
attempt to open //usr/x86_64-redhat-linux/lib/libtcl.so failed
attempt to open //usr/x86_64-redhat-linux/lib/libtcl.a failed
attempt to open //usr/local/lib/libtcl.so failed
attempt to open //usr/local/lib/libtcl.a failed
attempt to open //lib/libtcl.so failed
attempt to open //lib/libtcl.a failed
attempt to open //usr/lib/libtcl.so failed
attempt to open //usr/lib/libtcl.a failed
```

The target which is found by cmake is: `TCL_LIBRARY:FILEPATH=/usr/lib64/libtcl8.6.so`
which is a good and valid target, but LD does not pick this up.

Curiously, `/usr/lib/libtcl.so` _does_ exist, but it is a symlink to `./libtcl8.6.so` which doesn't exist. I think the link is broken and it should point to `../lib64/libtcl8.6.so`.

Alternatively, a symlink from `/usr/lib64/libtcl.so` to `/usr/lib64/libtcl8.6.so` would also work.

**Comment by rovinski** [comment]
Part of the issue may be that `/usr/lib64/libtcl.so` is installed by the yum package `tcl-devel` but `/usr/lib64/libtcl8.6.so` is installed by `tcl`.

**Comment by rovinski** [comment]
I think I found the issue: the `tcl` and `tcl-devel` packages are being installed from different repositories:
```
[root@583a99b0860f OpenROAD]# yum list installed tcl*
Updating Subscription Management repositories.
Unable to read consumer identity
Subscription Manager is operating in container mode.

This system is not registered with an entitlement server. You can use subscription-manager to register.

Installed Packages
tcl.x86_64                          1:8.6.8-2.el8         @ubi-8-baseos-rpms  <--
tcl-devel.i686                      1:8.6.8-2.el8         @@commandline  <--
tcl-tclreadline.x86_64              2.1.0-16.el8          @epel
tcl-tclreadline-devel.x86_64        2.1.0-16.el8          @epel
tcl-thread.x86_64                   2.8.8-1.el8           @epel
tcl-thread-devel.x86_64             2.8.8-1.el8           @epel
tcllib.noarch                       1.19-3.el8            @epel
```

This is likely causing some incompatibility. If I simply do:
```
yum remove tcl tcl-devel
yum install tcl tcl-devel
```
Then it reinstalls them from the same repo:
```
[root@583a99b0860f OpenROAD]# yum list installed tcl*
Updating Subscription Management repositories.
Unable to read consumer identity
Subscription Manager is operating in container mode.

This system is not registered with an entitlement server. You can use subscription-manager to register.

Installed Packages
tcl.x86_64                         1:8.6.8-2.el8         @baseos
tcl-devel.x86_64                   1:8.6.8-2.el8         @baseos
```
It then builds fine.
I will see if I can fix it and add it to #3286.

**Comment by rovinski** [comment]
Actually I don't think it had to do with different repositories, I think it had more to do with `tcl-devel` being installed directly from an RPM package. Either way, if we just use a different package, it seems to create the necessary symlink and then link correctly.
